common/src/lib.rs

```diff
@@ -171,11 +171,24 @@ pub enum Program {
     Terminated,
 }
 
-#[derive(Clone, Copy, Debug)]
-pub enum ProgramError {
-    CodeHashNotFound,
+#[derive(Clone, Debug, derive_more::Display)]
+pub enum CommonError {
```

Comment:
```suggestion
pub enum ProgramError {
```

I see no reason for renaming, cause it's still about the programs and its data only

Reply:
So, we can rename errors to ProgramError almost everywhere? Because everywhere errors about program execution. I can rename it to `Error`, but `ProgramError` is not uninformative

Reply:
Why the "Common" is more informative so?

Reply:
just has the same name with crate, I think is more comfortable to use outside crate